package NRTYArchive;


message TPartMetaInfo {
        enum TStage {
            OPENED = 1;
            BUILD = 2;
            CLOSED = 3;
            BROKEN = 4;
        }
    optional TStage Stage = 1 [default = BROKEN];
    optional uint64 DocsCount = 2 [default = 0];
    optional uint64 RemovedDocsCount = 3 [default = 0];
    optional string Compression = 4;
}

message THashMetaInfo {
    optional uint32 Version = 1;
    optional uint64 BucketsCount = 2 [default = 1000000];
    optional uint64 FreeCellIndex = 3 [default = 0];
}

message TArchiveMeta {
    optional float PopulationRate = 1[default = 0.6];
    optional float PartSizeDeviation = 2[default = 0.2];
    optional uint64 PartSizeLimit = 3[default = 1099511627776];
}
